### PR TITLE
Add a debconf plugin

### PR DESCRIPTION
**Pull request by nbraud**
This is my take on #282.

This lets you set debconf answers before extra packages are installed (but after the OS setup).

### CODE REVIEW

**Comment by andsens** [comment]
Nice! Good catch with the `additional_properties`.
The reason plugin configs are forced to be arrays is because of forwards compatibility. If you wanted to add a setting to a plugin that is configured with a single string, you'd have to break the manifests people are already using. By making it an object you can always just add another key alongside the existing one.

Also, what about this debconf parameter:
```
       -opackage, --owner=package
           Tell debconf what package the command it is running is a part of.
           This is necessary to get ownership of registered questions right,
           and to support unregister and purge commands properly.
```

Wouldn't you need a way to specify that?

**Comment by nbraud** [comment]
@andsens This plugin never calls `debconf(1)`.
The name of the package that owns a given question is always the first value on the line (see `debconf-set-selections(1)`).

Regarding forward-compatibility, I agree with the argument but I would expect package writers to choose on a case-by-case basis: here, I know that the scope of this plugin is feeding answers to `debconf(8)`, and the input format for `debconf(8)` is a plain string, so it seems like a reasonable choice.

**Comment by andsens** [comment]
> would expect package writers to choose on a case-by-case basis

True. It seems it would be fairly trivial to predict whether a plugin is never-ever going to get another parameter, especially if you want to keep it simple and rather a new plugin is written to handle other features.

> so it seems like a reasonable choice.

I agree. Merging :-)

File: bootstrapvz/plugins/debconf/schema.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+type: object
+properties:
+  plugins:
+    type: object
+    properties:
+      debconf:
+        name: Debconf selections to set
+        description: >-
+          This value should be an inline string in the
+          input format of debconf-set-selections(1).
```
Comment (andsens) [inline bootstrapvz/plugins/debconf/schema.yaml]: p.s.: I like this. It would be cool to pull `name` and `description` into the documentation somehow :-)

**Comment by nbraud** [comment]
Thanks a lot  :-)

**Pull request merged**